src/main/java/org/lookout_studios/meals_management_system/meals_management_system/RegistrationService.java

```diff
@@ -0,0 +1,125 @@
+    public ResponseEntity<String> registerUser(@RequestBody User user) throws Exception {
+        String userEmail = user.getEmail();
+        log.info(String.format(
+                "New registration request for user with email '%s'",
+                userEmail));
+        log.info(String.format(
+                "Checking if '%s' is a valid email",
+                userEmail));
+        if (!emailCheck(userEmail)) {
+            log.info(String.format("Email '%s' is invalid", userEmail));
+            return new ResponseEntity<String>(
+                    new ResponseBody(HttpStatus.BAD_REQUEST, invalidEmailMessage).getResponseBody(),
+                    HttpStatus.BAD_REQUEST);
+        }
+        if (!passwordCheck(user.getPassword())) {
+            log.info(String.format("Password provided by %s is invalid", userEmail));
+            return new ResponseEntity<String>(
+                    new ResponseBody(HttpStatus.BAD_REQUEST, invalidPasswordMessage).getResponseBody(),
+                    HttpStatus.BAD_REQUEST);
+        }
+        DatabaseService databaseService = new DatabaseService();
+        try {
+            log.info(String.format(
+                    "Checking if %s is already registered",
+                    userEmail));
+            boolean isRegistered = databaseService.isUserRegistered(userEmail);
+            if (isRegistered) {
+                log.info(String.format(
+                        "User with email %s is already registered",
+                        userEmail));
+                return new ResponseEntity<String>(
+                        new ResponseBody(
+                                HttpStatus.BAD_REQUEST,
+                                alreadyRegisteredMessage).getResponseBody(),
+                        HttpStatus.BAD_REQUEST);
+            }
+            log.info(String.format(
+                    "%s is not registered",
+                    userEmail));
+        } catch (Exception exception) {
+            throw exception;
+        }
+        log.info(String.format(
+                "Registering new user with email '%s'",
+                userEmail));
+        databaseService.registerNewUser(user);
+        // TO-DO: Send email with registration link containing the token
+        log.info(String.format(
+                "User with email %s registered successfully",
+                userEmail));
+        return new ResponseEntity<String>(
+                new ResponseBody(HttpStatus.OK).getResponseBody(), HttpStatus.OK);
+    }
+
+    /**
+     * Checks if a given string is a valid email address
+     * 
+     * @param email An email address to be checked
+     * @return True if a string is a valid email address, false if it isn't
+     */
+    public boolean emailCheck(String email) {
+        boolean match;
+        try {
+            Pattern validEmailPattern = Pattern.compile("^[\\w+-]+(\\.[\\w+-]+)*[\\.]?[a-zA-Z0-9]@([a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,6}$");
```

Comment:
Magic value is still present. Try defining it at the beginning of the class as static string with descriptive variable name and put it down here (as a variable).